### ISSUE TITLE
scanner fails, with another branches results

### ISSUE THREAD

**Issue by heidricha** [mc1arke/sonarqube-community-branch-plugin#239]
**Describe the bug**
sonar scanner failes on a (long-lived) branch. The quality-gate is reported OK, but scan still fails referring the results of the master branch's quality gate in the same project

**Expected behavior**
Another branch should not be involved at scanning a long-lived branch

**Software Versions**
 - SonarQube Version: 7.9.4.35981
 - Plugin Version: 1.3.2

**Additional context**
Scanner log:
 INFO: Executing post-job 'GitLab Commit Issue Publisher'
 INFO: Waiting quality gate to complete...
 INFO: Waiting quality gate to complete...
 INFO: Quality gate status: OK
 INFO: Reliability Rating : 1
 INFO: Security Rating : 1
 INFO: Maintainability Rating : 1
 INFO: Duplicated Lines (%) : 2.2
 INFO: Blocker Issues : 0
 INFO: Critical Issues : 0
 INFO: Line Coverage : 80.1
 INFO: Major Issues : 0
 INFO: ------------------------------------------------------------------------
 INFO: EXECUTION FAILURE
 INFO: ------------------------------------------------------------------------
 INFO: Total time: 1:05.251s
 INFO: Final Memory: 18M/70M
 INFO: ------------------------------------------------------------------------
 ERROR: Error during SonarScanner execution
 ERROR: Report status=failed, desc=SonarQube reported QualityGate is ok, with 8 ok, 24 issues, with 12 critical (fail) and 11 major and 1 minor
 ERROR: 
 ERROR: Re-run SonarScanner using the -X switch to enable full debug logging.

**Comment by mc1arke** [comment]
Could you provide the arguments you're passing to the scanner please?

**Comment by heidricha** [comment]
scanner runs in a container, execution is (gitlab CI job definition):

  - export SONAR_ARGS="
      -Dsonar.sources=/src 
      -Dsonar.gitlab.project_id=${CI_PROJECT_PATH}
      -Dsonar.gitlab.commit_sha=${CI_COMMIT_SHA}
      -Dsonar.gitlab.ref_name=${CI_COMMIT_REF_NAME}
      $(
        test -z "${CI_MERGE_REQUEST_ID}" && 
        printf " -Dsonar.branch.name=${CI_COMMIT_REF_NAME}" || 
        printf " -Dsonar.branch.name=${CI_MERGE_REQUEST_SOURCE_BRANCH_NAME} -Dsonar.branch.target=${CI_MERGE_REQUEST_TARGET_BRANCH_NAME}"
      )"
  - echo "$(id -un):x:$(id -u):$(id -g)::/src:/sbin/nologin" > docker/passwd
  - docker run --rm
    --add-host gitlab:${gitlab_host}
    --add-host sonar:${sonar_host}
    --add-host nexus:${nexus_host}
    -v $PWD/dashboard/dashboard/:/src
    -v $PWD/docker/passwd:/etc/passwd
    -w /src
    --user $(id -u):$(id -g)
    -e http_proxy=${http_proxy}
    -e https_proxy=${https_proxy}
    -e no_proxy="gitlab,sonar,nexus"
    ${DOCKER_REGISTRY}scom/dc-scanner
    bash -c "
      npm config set registry https://nexus/repository/npm/ &&
      npm install -D typescript && /usr/bin/sonar-scanner $SONAR_ARGS"

Actual arguments are not echoed, but I can update the code if necessary. Same job definition applies for merge requests, and long-lived branches.

```yaml
  only:
    refs:
    - merge_requests
    - develop
    - master
    - /^feature-release-v-/
    - /^feature-sonar-/
    changes:
    - dashboard/**/*

```

.sonar-project.properties:
sonar.projectKey=smartvision-web
sonar.projectName=smartvision-web
sonar.inclusions=/src/app/**/*.ts
sonar.exclusions=**/*.spec.ts
sonar.host.url=https://sonar/
sonar.login=5c8b8f9f9714d9568f9de41db140a7a2ac6f934c
sonar.cobertura.reportPath=coverage/cobertura-coverage.xml
sonar.typescript.lcov.reportPaths=coverage/lcov.info

**Comment by heidricha** [comment]
the actual problem arises on a normal (merge) commit, not a merge request

**Comment by mc1arke** [comment]
I'd need you to provide the arguments you're actually generating as a result of your script otherwise I'm making assumptions about the arguments you're passing in.

**Comment by heidricha** [comment]
```SONAR_ARGS= -Dsonar.projectVersion=3.0.0 -Dsonar.sources=/src -Dsonar.gitlab.project_id=smartvision/SV-SmartVision -Dsonar.gitlab.commit_sha=63fce12f885dac2973e42ecb7f5af438790104c9 -Dsonar.gitlab.ref_name=feature-SVI-0-sonar-Adds-DBG-for-sonar  -Dsonar.branch.name=feature-SVI-0-sonar-Adds-DBG-for-sonar```

**Comment by 000-010-000** [comment]
Hello. 
The same problem. 

**SonarQube server** 7.9.1
**SonarQube Scanner** 4.1.0.1829
**Plugin Version**: 1.3.2 (1.1.1 as well)

**Args**:   -X -Dsonar.verbose=true  -Dsonar.qualitygate.wait=true -Dproject.settings=./sonar-project.properties -Dsonar.login=${SONAR_LOGIN} -Dsonar.branch.name=${CI_COMMIT_REF_NAME} -Dsonar.gitlab.project_id=${CI_PROJECT_ID} -Dsonar.gitlab.commit_sha=${CI_COMMIT_SHA} -Dsonar.gitlab.ref_name=${CI_COMMIT_REF_NAME} -Dsonar.analysis.joburl=${CI_JOB_URL}

**sonar-project.properties:** 
sonar.projectKey=Taskmanager
sonar.projectName=Task Manager
sonar.host.url=https://sonarqube.company.domain
sonar.sourceEncoding=UTF-8
sonar.sources=.
sonar.working.directory=.sonarprojectTaskmanager


[Full debug log](https://github.com/mc1arke/sonarqube-community-branch-plugin/files/5290892/raw_log.log)

**Comment by lucasoares** [comment]
I got this issue once but I honestly don't know how to reproduce it.

**Comment by mc1arke** [comment]
Is this still happening with the latest Sonarqube LTS version?
